boards/BMS/Src/App/App_Accumulator.c

```diff
@@ -100,30 +103,105 @@ void App_Accumulator_Destroy(struct Accumulator *accumulator)
     free(accumulator);
 }
 
+void App_Accumulator_InitRunOnEntry(const struct Accumulator *const accumulator)
+{
+    // Configure the cell monitoring chips. Disable discharge at startup
+    accumulator->config_monitoring_chip();
+}
+
 bool App_Accumulator_HasCommunicationError(const struct Accumulator *const accumulator)
 {
     return accumulator->num_comm_tries >= MAX_NUM_COMM_TRIES;
 }
 
-float App_Accumulator_GetPackVoltage(struct Accumulator *accumulator)
+void App_Accumulator_ReadCellVoltage(struct Accumulator *accumulator)
 {
-    return accumulator->get_pack_voltage();
+    UPDATE_PEC15_ERROR_COUNT(accumulator->read_cell_voltages(accumulator->se_voltages), accumulator->num_comm_tries)
 }
 
-void App_Accumulator_InitRunOnEntry(const struct Accumulator *const accumulator)
+void App_Accumlator_CalcVoltageStats(struct Accumulator *accumulator)
 {
-    // Configure the cell monitoring chips. Disable discharge at startup
-    accumulator->config_monitoring_chip();
+    struct VoltageStats temp_voltage_stats = { .min = { .segment = 0U, .series_elem = 0U, .voltage = UINT16_MAX },
+                                               .max = { .segment = 0U, .series_elem = 0U, .voltage = 0x0000U },
+                                               .segment_avg = { 0U },
+                                               .segment     = { 0U },
+                                               .pack_avg    = 0U,
+                                               .pack        = 0U };
+
+    // find the min and max voltages
```

Comment:
Nit: should stay consistent with the capitalization of first character in these comments (more below)
```suggestion
    // Find the min and max voltages
```